File: 코딩테스트 고득점 Kit/스택과큐/프로세스/프로세스.py
from collections import deque


def solution(priorities, location):
    answer = 0
    dq = deque([])

    for i in range(len(priorities)):
        dq.append((priorities[i], i))

    while dq:
        curVal, curIdx = dq.popleft()
        # print(curIdx, curVal)
        if dq and curVal < max(dq)[0]:
            dq.append((curVal, curIdx))
        else:
            answer += 1
            if curIdx == location:
                break

    return answer

File: 코딩테스트 고득점 Kit/스택과큐/프로세스/프로세스.swift
import Foundation

func solution(_ priorities:[Int], _ location:Int) -> Int {
    var answer:Int = 0
    var pq:[Int] = []
    var que:[(Int, Int)] = []
    
    for i in 0..<priorities.count {
        que.append((i, priorities[i]))
        pq.append(priorities[i])
    }
    
    pq.sort(by: >)
    
    while !pq.isEmpty {
        let cur = que.removeFirst()
        
        if cur.1 == pq.first! {
            answer += 1
            pq.removeFirst()
            
            if cur.0 == location {
                break
            }
        } else {
            que.append(cur)
        }
    }
    
    return answer
}